src/attack_flow_builder/src/assets/builder.config.validator.ts

```diff
@@ -213,10 +213,10 @@ class AttackFlowValidator extends DiagramValidator {
                     this.addError(id, "Invalid MIME Type.");
                 }
                 // Validate Payload Bin, URL, and Hashes
-                if (payloadBin?.isDefined() && (url?.isDefined() || hashes?.isDefined()) ||
-                   (url?.isDefined() && !hashes?.isDefined()) ||
-                   (hashes?.isDefined() && !url?.isDefined())) {
-                        this.addError(id, "An artifact must have either a payload or a URL + hash, but not both.");
+                if (payloadBin?.isDefined() === url?.isDefined()) {
```

Comment:
This `if` statement fails if both `payload_bin` and `url` are blank, but according to the [STIX Documentation](https://docs.oasis-open.org/cti/stix/v2.1/os/stix-v2.1-os.html#_4jegwl6ojbes), both fields are optional.

Reply:
Fix applied.